server/mongo_data_access.go

```diff
@@ -13,12 +13,61 @@ import (
 )
 
 // NewMongoDataAccessLayer returns an implementation of DataAccessLayer that is backed by a Mongo database
-func NewMongoDataAccessLayer(db *mgo.Database) DataAccessLayer {
-	return &mongoDataAccessLayer{Database: db}
+func NewMongoDataAccessLayer(db *mgo.Database, interceptors map[string]InterceptorList) DataAccessLayer {
+	return &mongoDataAccessLayer{
+		Database:     db,
+		Interceptors: interceptors,
+	}
 }
 
 type mongoDataAccessLayer struct {
-	Database *mgo.Database
+	Database     *mgo.Database
+	Interceptors map[string]InterceptorList
+}
+
+// An interceptor executes a function on a specified resource type immediately BEFORE
```

Comment:
So this is a weird Go thing, but when comment Go types or functions, the first word in the comment should be the type's (or function's) name.  In this case, "Interceptor wraps a function that should be executed on a specified resource type...".

Same goes for most of your comments (but KUDOS to you for commenting at all!).